### PR TITLE
Add activity log alert commands.

### PR DESCRIPTION
**Pull request by troydai**
## Commands

```
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
> az monitor activity-log alert -h                                                                           
                                                                                                             
Group                                                                                                        
    az monitor activity-log alert: Manage activity log alerts.                                               
                                                                                                             
Subgroups:                                                                                                   
    action: Manage action groups for activity log alerts.                                                    
    scope                                                                                                    
                                                                                                             
Commands:                                                                                                    
    create: Create a default activity log alert.                                                             
    delete: Delete an activity log alert.                                                                    
    list  : List activity log alerts under a resource group or the current subscription.                     
    show  : Get an activity log alert.                                                                       
    update: Update the details of this activity log alert.                                                   
                                                                                                             
                                                                                                             
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
> az monitor activity-log alert -h                                                                           
                                                                                                             
Group                                                                                                        
    az monitor activity-log alert: Manage activity log alerts.                                               
                                                                                                             
Subgroups:                                                                                                   
    action: Manage action groups for activity log alerts.                                                    
    scope : Manage scopes for activity log alerts.                                                           
                                                                                                             
Commands:                                                                                                    
    create: Create a default activity log alert.                                                             
    delete: Delete an activity log alert.                                                                    
    list  : List activity log alerts under a resource group or the current subscription.                     
    show  : Get an activity log alert.                                                                       
    update: Update the details of this activity log alert.                                                   
                                                                                                             
                                                                                                             
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
> az monitor activity-log alert action -h                                                                    
                                                                                                             
Group                                                                                                        
    az monitor activity-log alert action: Manage action groups for activity log alerts.                      
                                                                                                             
Commands:                                                                                                    
    add   : Add action groups to this activity log alert.                                                    
    remove: Remove action groups from this activity log alert.                                               
                                                                                                             
                                                                                                             
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
> az monitor activity-log alert scope -h                                                                     
                                                                                                             
Group                                                                                                        
    az monitor activity-log alert scope: Manage scopes for activity log alerts.                              
                                                                                                             
Commands:                                                                                                    
    add   : Add scopes to this activity log alert.                                                           
    remove: Removes scopes from this activity log alert.                                                     
                                                                                                             
                                                                                                             
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
> az monitor activity-log alert update -h                                                                    
                                                                                                             
Command                                                                                                      
    az monitor activity-log alert update: Update the details of this activity log alert.                     
                                                                                                             
Arguments                                                                                                    
    --condition -c     : A condition expression represents the condition that will cause the alert           
                         to activate. The format is FIELD=VALUE[ and FILED=VALUE...]. The possible           
                         values for the field are 'resourceId', 'category', 'caller', 'level',               
                         'operationName', 'resourceGroup', 'resourceProvider', 'status',                     
                         'subStatus', 'resourceType', or anything beginning with 'properties.'.              
    --description      : A description of this activity log alert.                                           
    --enabled          : Allowed values: false, true.                                                        
    --tags             : Space separated tags in 'key[=value]' format. Use "" to clear existing              
                         tags.                                                                               
                                                                                                             
Resource Id Arguments                                                                                        
    --ids              : One or more resource IDs (space delimited). If provided, no other 'Resource         
                         Id' arguments should be specified.                                                  
    --name -n          : The name of the activity log alert.                                                 
    --resource-group -g: Name of resource group. You can configure the default group using `az               
                         configure --defaults group=<name>`.                                                 
                                                                                                             
Generic Update Arguments                                                                                     
    --add              : Add an object to a list of objects by specifying a path and key value               
                         pairs.  Example: --add property.listProperty <key=value, string or JSON             
                         string>.                                                                            
    --remove           : Remove a property or an element from a list.  Example: --remove                     
                         property.list <indexToRemove> OR --remove propertyToRemove.                         
    --set              : Update an object by specifying a property path and value to set.  Example:          
                         --set property1.property2=<value>.                                                  
                                                                                                             
Global Arguments                                                                                             
    --debug            : Increase logging verbosity to show all debug logs.                                  
    --help -h          : Show this help message and exit.                                                    
    --output -o        : Output format.  Allowed values: json, jsonc, table, tsv.  Default: json.            
    --query            : JMESPath query string. See http://jmespath.org/ for more information and            
                         examples.                                                                           
    --verbose          : Increase logging verbosity. Use --debug for full debug logs.                        
                                                                                                             
(az36) troyd@TROY-THINKPAD25 C:\Users\troyd\Code\azure-cli                                                   
>                                                                                                            
```

### CODE REVIEW

**Comment by tjprescott** [comment]
From a very high level, it seems like "condtion" and "action" should be subgroups with collection-type commands (not unlike app gateway or load balancer subresources) rather than a single command with "command-like" arguments (add/remove-foo)

**Comment by troydai** [comment]
The current pattern is in line with the action group. It offers a few benefits (other than the disadvantage of crowded parameters):
* Allow the user to change multiple conditions of one alert in one command while still allow them to do it in a fashion of changing condition individually.
* More performant than individual add/remove command since actions can be bind together.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 1]
```diff
@@ -261,3 +261,139 @@
...
+          short-summary: Disable the activity log alert after it is created.
+        - name: --description
+          short-summary: A description of this activity log alert
+        - name: --condition -c
+          short-summary: Add condition. Accept space separated conditions in 'FIELD=VALUE' format. The possible values
+                         for the field are (case-insensitive) 'resourceId', 'category', 'caller', 'level',
+                         'operationName', 'resourceGroup', 'resourceProvider', 'status', 'subStatus', 'resourceType',
+                         or anything beginning with 'properties.'. If a condition with the same field exists, its value
+                         will be updated.
+        - name: --action -a
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: This should be `--action-group`.  Why doesn't it accept names or IDs?

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 2]
```diff
@@ -261,3 +261,139 @@
...
+    type: command
+    short-summary: Create a default activity log alert
+    long-summary: This command will create a default activity log with one condition which compares if the activities
+                  logs 'category' field equals to 'ServiceHealth'. The newly created activity log alert does not have
+                  any action groups attached to it.
+    parameters:
+        - name: --name -n
+          short-summary: Name of the activity log alerts
+        - name: --scope -s
+          short-summary: A list of resourceIds that will be used as prefixes. The alert will only apply to activityLogs
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: I thought these were just string prefixes?

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 3]
```diff
@@ -261,3 +261,139 @@
...
+                         with resourceIds that fall under one of these prefixes. This list must include at least one
+                         item.
+        - name: --disable
+          short-summary: Disable the activity log alert after it is created.
+        - name: --description
+          short-summary: A description of this activity log alert
+        - name: --condition -c
+          short-summary: Add condition. Accept space separated conditions in 'FIELD=VALUE' format. The possible values
+                         for the field are (case-insensitive) 'resourceId', 'category', 'caller', 'level',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: Condition only test for equality?  What about other conditional operations?  Also, is there a difference between:

`-c A=B C=D` and `-c A=B -c C=D`?  As I user I would interpret these as, for the first one A=B AND C=D and then interpret the second as A=B  OR C=D.  But I don't know how/if the service interprets these, and the help is not clear.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 4]
```diff
@@ -261,3 +261,139 @@
...
+                         'operationName', 'resourceGroup', 'resourceProvider', 'status', 'subStatus', 'resourceType',
+                         or anything beginning with 'properties.'. If a condition with the same field exists, its value
+                         will be updated.
+        - name: --action -a
+          short-summary: Add action group. Accept space separated resource id of the action groups to be added. The
+                         invalid resource id is ignored.
+        - name: --webhook-properties -w
+          short-summary: Space separated web hook properties in 'key[=value]' format. These properties will be
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: This is confusing to me. Examples would help tremendously.  Are these basically the args that are sent to the webhook as query parameters or as part of the body?  Are these only static values?

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 5]
```diff
@@ -261,3 +261,139 @@
...
+                         associated with the action groups added in this command. For any webhook receiver in these
+                         action group, these data are appended to the webhook payload. To attach different webhook
+                         properties to different action groups, add the action groups in separate update-action
+                         commands.
+"""
+
+helps['monitor activity-log alert enable'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: This should be folded into the update command.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 6]
```diff
@@ -261,3 +261,139 @@
...
+    type: command
+    short-summary: Enable this activity log alert
+"""
+
+helps['monitor activity-log alert disable'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: This should be folded into the update command.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 7]
```diff
@@ -261,3 +261,139 @@
...
+    parameters:
+        - name: --name -n
+          short-summary: Name of the activity log alerts
+        - name: --add-condition -a
+          short-summary: Add condition. Accept space separated conditions in 'FIELD=VALUE' format. The possible values
+                         for the field are (case-insensitive) 'resourceId', 'category', 'caller', 'level',
+                         'operationName', 'resourceGroup', 'resourceProvider', 'status', 'subStatus', 'resourceType',
+                         or anything beginning with 'properties.'. If a condition with the same field exists, its value
+                         will be updated.
+        - name: --remove-condition -r
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: These should be folded into the update command.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 8]
```diff
@@ -261,3 +261,139 @@
...
+          short-summary: Remove conditions. Accept space separated conditions in 'FIELD=VALUE' format. The removing is
+                         executed before adding. Some condition, such as 'category', is required for this alert to be
+                         operational, therefore remove without replace with a new one will cause service error.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: Does this go through all conditions and remove all instances of this key=value pair?

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 9]
```diff
@@ -261,3 +261,139 @@
...
+        - name: --reset
+          short-summary: Remove all the existing conditions before add new conditions. When --reset is given, the values
+                         of --remove-condition will be ignored.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: Alternatively, you could simply accept `--remove-condition *` to remove all conditions.  Intuitive and minimizes arguments.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 10]
```diff
@@ -261,3 +261,139 @@
...
+helps['monitor activity-log alert update-action'] = """
+    type: command
+    short-summary: Update the action groups of this activity log alert.
+    parameters:
+        - name: --name -n
+          short-summary: Name of the activity log alerts
+        - name: --add-action -a
+          short-summary: Add action group. Accept space separated resource id of the action groups to be added. The
+                         invalid resource id is ignored.
+        - name: --remove-action -r
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: These should be folded into update.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 11]
```diff
@@ -261,3 +261,139 @@
...
+helps['monitor activity-log alert update-scope'] = """
+    type: command
+    short-summary: Update the scopes of this activity log alert.
+    parameters:
+        - name: --name -n
+          short-summary: Name of the activity log alerts
+        - name: --add-scope -a
+          short-summary: A list of resourceIds that will be used as prefixes. The alert will only apply to activityLogs
+                         with resourceIds that fall under one of these prefixes.
+        - name: --remove-scope -r
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: I didn't see this when I ran the help (is the command actually registered?) but this should be folded into update.

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 12]
```diff
@@ -261,3 +261,139 @@
...
+          short-summary: A list of the scopes to be removed.
+        - name: --reset
+          short-summary: Remove all the existing scopes before add new scopes. When --reset is given, the values of
+                         --remove-scope will be ignored.
+"""
+
+helps['monitor activity-log alert update'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: Most of the commands in this group should be folded into this command. This command should also be a generic update.

**Comment by tjprescott** [comment]
To summarize our offline discussion:

- [ ] make update generic, with --enable, --condition, --tags --description as arguments.
- [ ] Remove enable/disable commands and the extra update commands
- [ ] Add `action-group` subcommand with `add/remove/update` (possibly clear, up to you) commands.
- [ ] Add `scope` subcommand with `add/remove` commands.
- [ ] Clarify that --condition all values are ANDs (or parse a syntax)
- [ ] Examples! 💃

**Comment by troydai** [comment]
@tjprescott a few more things to be added, but the major change is here. take another look.

**Comment by tjprescott** [comment]
Added a couple more things to that list, but it's looking much better to me 😸

**Comment by troydai** [comment]
Oh, no the list is expanding.

**Comment by troydai** [comment]
Updated!

File: src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py [hunk 13]
```diff
@@ -261,3 +261,183 @@
...
+        - name: --description
+          short-summary: A description of this activity log alert
+        - name: --condition -c
+          short-summary: A condition expression represents the condition that will cause the alert to activate. The
+                         format is FIELD=VALUE[ and FILED=VALUE...]. The possible values for the field are 'resourceId',
+                         'category', 'caller', 'level', 'operationName', 'resourceGroup', 'resourceProvider', 'status',
+                         'subStatus', 'resourceType', or anything beginning with 'properties.'.
+        - name: --action-group -a
+          short-summary: Add action group. Accept space separated action groups identifiers. The identify can be the
+                         action group's name (if it is in the same resource group as the alert) or its resource id.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: > (if it is in the same resource group as the alert)

This is the standard behavior of the CLI so you shouldn't need to call it out.

Comment (troydai) [reply src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_help.py]: Cool.

**Comment by troydai** [comment]
Waiting for CI.

**Pull request merged**